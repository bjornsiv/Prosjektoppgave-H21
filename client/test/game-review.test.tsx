import * as React from 'react';
import mockReviewService from '../__mocks__/mock-review-service';
import mockGameService from '../__mocks__/mock-game-service';
import renderer from 'react-test-renderer';
import GameReview from '../src/new-review';

jest.mock('../src/game-service', () => {
  return new mockGameService();
});
jest.mock('../src/review-service', () => {
  return new mockReviewService();
});

describe('Game review component tests', () => {
  test('GameReview renders correctly with snapshot', () => {
    const tree = renderer
      .create(<GameReview match={{ params: { id: 1 } }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});