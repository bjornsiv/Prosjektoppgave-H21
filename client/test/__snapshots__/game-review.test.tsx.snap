// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game review component tests GameReview renders correctly with snapshot 1`] = `
Array [
  <div
    className="card rounded"
  >
    <div
      className="card-body"
    >
      <p
        className="card-title"
      >
        Add a new review for 
      </p>
      <div
        className="card-text"
      >
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              <span>
                <label
                  className="form col-form-label form-display"
                >
                  Title:
                </label>
              </span>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              <input
                className="form-input form form-control"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              <span>
                <label
                  className="form col-form-label form-display"
                >
                  Description:
                </label>
              </span>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              <textarea
                className="form-control"
                onChange={[Function]}
                rows={10}
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              Score:
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              <div
                aria-label="Game rating"
                className="custom-star-rating btn-group"
                role="group"
                style={
                  Object {
                    "backgroundSize": "100%",
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <button
                  className="star-btn btn btn-outline-secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "borderColor": "transparent",
                    }
                  }
                  title="0"
                  type="button"
                >
                  <svg
                    className="star"
                    fill="yellow"
                    height="16"
                    style={
                      Object {
                        "fontSize": 28,
                      }
                    }
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.522-3.356c.33-.314.16-.888-.282-.95l-4.898-.696L8.465.792a.513.513 0 0 0-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767-3.686 1.894.694-3.957a.565.565 0 0 0-.163-.505L1.71 6.745l4.052-.576a.525.525 0 0 0 .393-.288L8 2.223l1.847 3.658a.525.525 0 0 0 .393.288l4.052.575-2.906 2.77a.565.565 0 0 0-.163.506l.694 3.957-3.686-1.894a.503.503 0 0 0-.461 0z"
                    />
                  </svg>
                </button>
                <button
                  className="star-btn btn btn-outline-secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "borderColor": "transparent",
                    }
                  }
                  title="0"
                  type="button"
                >
                  <svg
                    className="star"
                    fill="yellow"
                    height="16"
                    style={
                      Object {
                        "fontSize": 28,
                      }
                    }
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.522-3.356c.33-.314.16-.888-.282-.95l-4.898-.696L8.465.792a.513.513 0 0 0-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767-3.686 1.894.694-3.957a.565.565 0 0 0-.163-.505L1.71 6.745l4.052-.576a.525.525 0 0 0 .393-.288L8 2.223l1.847 3.658a.525.525 0 0 0 .393.288l4.052.575-2.906 2.77a.565.565 0 0 0-.163.506l.694 3.957-3.686-1.894a.503.503 0 0 0-.461 0z"
                    />
                  </svg>
                </button>
                <button
                  className="star-btn btn btn-outline-secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "borderColor": "transparent",
                    }
                  }
                  title="0"
                  type="button"
                >
                  <svg
                    className="star"
                    fill="yellow"
                    height="16"
                    style={
                      Object {
                        "fontSize": 28,
                      }
                    }
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.522-3.356c.33-.314.16-.888-.282-.95l-4.898-.696L8.465.792a.513.513 0 0 0-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767-3.686 1.894.694-3.957a.565.565 0 0 0-.163-.505L1.71 6.745l4.052-.576a.525.525 0 0 0 .393-.288L8 2.223l1.847 3.658a.525.525 0 0 0 .393.288l4.052.575-2.906 2.77a.565.565 0 0 0-.163.506l.694 3.957-3.686-1.894a.503.503 0 0 0-.461 0z"
                    />
                  </svg>
                </button>
                <button
                  className="star-btn btn btn-outline-secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "borderColor": "transparent",
                    }
                  }
                  title="0"
                  type="button"
                >
                  <svg
                    className="star"
                    fill="yellow"
                    height="16"
                    style={
                      Object {
                        "fontSize": 28,
                      }
                    }
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.522-3.356c.33-.314.16-.888-.282-.95l-4.898-.696L8.465.792a.513.513 0 0 0-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767-3.686 1.894.694-3.957a.565.565 0 0 0-.163-.505L1.71 6.745l4.052-.576a.525.525 0 0 0 .393-.288L8 2.223l1.847 3.658a.525.525 0 0 0 .393.288l4.052.575-2.906 2.77a.565.565 0 0 0-.163.506l.694 3.957-3.686-1.894a.503.503 0 0 0-.461 0z"
                    />
                  </svg>
                </button>
                <button
                  className="star-btn btn btn-outline-secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "borderColor": "transparent",
                    }
                  }
                  title="0"
                  type="button"
                >
                  <svg
                    className="star"
                    fill="yellow"
                    height="16"
                    style={
                      Object {
                        "fontSize": 28,
                      }
                    }
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.522-3.356c.33-.314.16-.888-.282-.95l-4.898-.696L8.465.792a.513.513 0 0 0-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767-3.686 1.894.694-3.957a.565.565 0 0 0-.163-.505L1.71 6.745l4.052-.576a.525.525 0 0 0 .393-.288L8 2.223l1.847 3.658a.525.525 0 0 0 .393.288l4.052.575-2.906 2.77a.565.565 0 0 0-.163.506l.694 3.957-3.686-1.894a.503.503 0 0 0-.461 0z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="float-start"
      >
        <button
          className="btn btn-dark button-darker"
          onClick={[Function]}
          type="button"
        >
          Publish
        </button>
      </div>
    </div>
  </div>,
]
`;
